include/palotasb/static_vector.hpp

```diff
@@ -363,22 +344,26 @@ struct static_vector {
         // Need mutable iterator to change items. Cast is legal in non-const
         // methos.
         iterator mut_pos = const_cast<iterator>(pos);
-        // move_backward is recommended when the end of the target range is
-        // outside the input range, last element is moved first
-        std::move_backward(mut_pos, end(), end() + 1);
-        // Construct value, do not assign nonexistent
-        new (mut_pos) value_type(std::forward<CtorArgs>(args)...);
+
+        // Insert the item at the end and then rotate to put into the correct
+        // place
+        new (end()) value_type(std::forward<CtorArgs>(args)...);
         m_size++;
+
+        std::rotate(mut_pos, end()-1, end());
+
         return mut_pos;
     }
 
     // Erase element at `pos`
     // TODO docs
     iterator erase(const_iterator pos) {
         iterator mut_pos = const_cast<iterator>(pos);
-        mut_pos->~value_type();
-        // move forward, starting from mut_pos and going towards end()
-        std::move(mut_pos + 1, end(), mut_pos);
+
+        // Rotate the array first and then call the destructor
+        std::rotate(mut_pos, mut_pos + 1, end());
+
+        (end()-1)->~value_type();
```

Comment:
```suggestion
        back().~value_type();
```

I think this helper could be used here.